lib/task_helper.rb

```diff
@@ -0,0 +1,45 @@
+require 'json'
+
+class TaskHelper
+  class Error < RuntimeError
+    attr_reader :kind, :details, :issue_code
+
+    def initialize(msg, kind, details = nil)
+      super(msg)
+      @kind = kind
+      @issue_code = issue_code
+      @details = details || {}
+    end
+
+    def to_h
+      { 'kind' =>  kind,
+        'msg' => message,
+        'details' => details }
+    end
+  end
+
+  def task(params = {})
+    msg = 'The task author must implement the `task` method in the task'
+    raise TaskHelper::Error.new(msg,
+                                'tasklib/not-implemented')
+  end
+
+  def self.run
+    params = JSON.parse(STDIN.read)
+
+    # This method accepts a hash of parameters to run the task, then executes
+    # the task. Unhandled errors are caught and turned into an error result.
+    # @param [Hash] params A hash of params for the task
+    # @return [Hash] The result of the task
+    result = begin self.task(params)
```

Comment:
This fails rspec locally with `undefined method `task' for ErrorTask:Class>` and I'm not sure why...